GUI.py

```diff
@@ -0,0 +1,647 @@
+        self.mainSizer.Add(self.controlSizer, 0, wx.RIGHT, 5)
+        self.SetSizer(self.mainSizer)
+        self.Fit()
+
+
+class MyPanel(wx.Panel):
+    """Custom panel class."""
+
+    def __init__(self, parent: MyFrame) -> None:
+        super(MyPanel, self).__init__(parent=parent)
+
+    def show_buttons(self) -> None:
+        hbox = wx.BoxSizer(wx.VERTICAL)
+        self.btn = wx.Button(self, label="Open Text File")
+        self.btn.Bind(wx.EVT_BUTTON, self.on_open)
+
+        # -----------------
+        self.cwd = os.path.join(
+            os.path.abspath(os.path.dirname(__file__)), "Refraction_index"
+        )
+
+        refl_files = get_files_list(self.cwd)
+        self.wvlngh, self.n_coef = refection_coef_read(
+            os.path.join(self.cwd, refl_files[1])
+        )
+        cb = wx.ComboBox(
+            self, choices=refl_files, style=wx.CB_READONLY, value=refl_files[1]
+        )
+        cb.Bind(wx.EVT_COMBOBOX, self.on_select)
+        self.st = wx.StaticText(self, label="")
+        self.st.SetLabel("Refractive index file:")
+        # -----------------
+
+        self.auto = wx.Button(self, label="Auto GaAs depth")
+        self.auto.Enable(False)
+        self.auto.Bind(wx.EVT_BUTTON, self.calc_auto)
+        # ------------------
+
+        self.calc_btn = wx.Button(self, label="Calculate depth")
+        self.calc_btn.Enable(False)
+        self.calc_btn.Bind(wx.EVT_BUTTON, self.calc_choice)
+        # ------------------
+
+        self.chk_box = wx.CheckBox(self, label="Full Report")
+
+        hbox.Add(self.btn, 0, wx.CENTER, 5)
+        hbox.Add(self.auto, 0, wx.CENTER, 5)
+        hbox.Add(self.calc_btn, 0, wx.CENTER | wx.BOTTOM, 10)
+        hbox.Add(self.st, 0, wx.EXPAND | wx.ALIGN_CENTER_HORIZONTAL | wx.TOP, 5)
+        hbox.Add(cb, 0, wx.EXPAND | wx.ALIGN_CENTER_HORIZONTAL | wx.ALL, 1)
+        hbox.Add(self.chk_box, 0, wx.EXPAND | wx.ALIGN_CENTER_HORIZONTAL | wx.TOP, 5)
+
+        self.SetSizer(hbox)
+
+    def on_select(self, event: wx.Event) -> None:
+        i = event.GetString()
+        self.wvlngh, self.n_coef = refection_coef_read(os.path.join(self.cwd, i))
+
+    def on_open(self, event: wx.Event) -> None:
+
+        wildcard = "TXT files (*.txt)|*.txt"
+        dialog = wx.FileDialog(
+            self,
+            "Open Text Files",
+            wildcard=wildcard,
+            style=wx.FD_OPEN | wx.FD_FILE_MUST_EXIST,
+        )
+
+        if dialog.ShowModal() == wx.ID_CANCEL:
+            return
+
+        path = dialog.GetPath()
+
+        if os.path.exists(path):
+            self.data = np.loadtxt(path)
+
+        self.calc_btn.Enable(False)
+
+        draw_data(self.graph, self.data[:, 0], self.data[:, 1], name="Spectrum")
+
+        self.auto.Enable(True)
+
+        if self.graphs.IsShown():
+            self.graphs.Hide()
+            self.graph.Show()
+
+        self.Parent.Layout()
+        self.Parent.Fit()
+
+    def calc_choice(self, event: wx.Event) -> None:
```

Comment:
event - not used parameter

Reply:
No, this function is event-dependent: you press the button, and the code runs. So 2 positional arguments are given to this function.